content/opt/setup/30-locales

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+if [[ ! -f /etc/locale.gen.dist ]]
+then
+   cp /etc/locale.gen /etc/locale.gen.dist
+fi
+
+echo "BUILD_LOCALES: $BUILD_LOCALES"
+typeset -i m=$(echo $BUILD_LOCALES | awk -F ',' '{print NF}')
+typeset -i n=1
+while [[ $n -le $m ]]
+do
+   b_locale=$(echo $BUILD_LOCALES | awk -F ',' "{print \$$n}")
+   cat /etc/locale.gen |grep -q "^$b_locale"
+   if [[ $? -ne 0 ]]
+   then
```

Comment:
The if clause actually executes a command and then tests for its exit code. If the exit code is 0 (==true), it'll execute the codeblock the result else not.

So the `[[` (or `[` depending on your shell) is called the test function, which is actually a just regular command, getting fed with the logical operators and evaluating them. The "matching" `]]` isn't actually a bash syntactic token, it's just an argument to the command and this command can verify that it received all arguments.

So, extratesting for `$? -ne 0` is superfluous, as you just could use `if grep -q "^$b_locale" /etc/locale.gen; then` and three lines are shrunken to a single one.